src/goto-programs/generate_function_bodies.cpp

```diff
@@ -125,7 +126,7 @@ class assert_false_then_assume_false_generate_function_bodiest
       auto instruction = function.body.add_instruction();
       instruction->function = function_name;
       instruction->source_location = function_symbol.location;
-      instruction->source_location.set_function(function_name);
+      instruction->source_location.set_function(function_symbol.display_name());
```

Comment:
Using display_name is a good idea in principle, but using it only here will result in inconsistent output. More importantly, there are many usages of `source_location.get_function()` that expect it to return a symbol identifier. These would need to be fixed first.

Reply:
That's scary - would you be able to point me to any of those `get_function()` uses so that I have a starting point?

Reply:
Most of the occurrences use the function for uniqueness of names, e.g. 
https://github.com/diffblue/cbmc/blob/706e3915232880c4c68befd4cd60ffb40a513d86/src/goto-programs/set_properties.cpp#L58 

This seems unnecessary because we could take the function from the instruction (same for XML):
https://github.com/diffblue/cbmc/blob/7f3ba3005dd0b77364de5e613242ae4680220d6d/src/goto-programs/json_goto_trace.cpp#L191

I doubt this would guarantee uniqueness of property names even at the moment unless pretty names are as unique as identifiers (this code is probably duplicated in many places):
https://github.com/diffblue/cbmc/blob/7f3ba3005dd0b77364de5e613242ae4680220d6d/src/goto-programs/json_goto_trace.cpp#L47

There are a couple of this sort that I am to blame for:
https://github.com/diffblue/cbmc/blob/c6beb6869464d22da6891cb52e4f6ee681306c15/src/ansi-c/c_nondet_symbol_factory.cpp#L39

...

Reply:
I am now refraining from using the `display_name`. The PR is now all about just setting the function name.